pkg/virt-handler/vm.go

```diff
@@ -286,21 +286,22 @@ func handleDomainNotifyPipe(domainPipeStopChan chan struct{}, ln net.Listener, v
 	// Listen for new connections,
 	// Close listener and exit when stop encountered
 	go func(vmi *v1.VirtualMachineInstance, ln net.Listener, domainPipeStopChan chan struct{}) {
+		go func() {
+			<-domainPipeStopChan
+			log.Log.Object(vmi).Infof("closing notify pipe listener for vmi")
+			ln.Close()
+		}()
 		for {
-			select {
-			case <-domainPipeStopChan:
-				log.Log.Object(vmi).Infof("closing notify pipe listener for vmi")
-				ln.Close()
-				return
-			default:
-				fd, err := ln.Accept()
-				if err != nil {
-					log.Log.Reason(err).Error("Domain pipe accept error encountered.")
-					// keep listening until stop invoked
-					time.Sleep(1)
-				} else {
-					fdChan <- fd
+			fd, err := ln.Accept()
+			if err != nil {
+				if goerror.Is(err, net.ErrClosed) {
+					return
 				}
+				log.Log.Reason(err).Error("Domain pipe accept error encountered.")
+				// keep listening until stop invoked
+				time.Sleep(1)
```

Comment:
@davidvossel @rmohr do you know if we really want to sleep for 1 Nanosecond here? I think this should probably be `1 * time.Millisecond` or more.

Reply:
use `1 * time.Second`. that was likely the original intent.